pkg/operator/configobservation/proxy/observe_proxy.go

```diff
@@ -0,0 +1,73 @@
+package proxy
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+func ObserveProxyConfig(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	const proxyPath = "proxy"
+
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedProxyConfig := map[string]interface{}{}
+
+	// grab the current Proxy config to later check whether it was updated
+	currentProxyMap, _, err := unstructured.NestedStringMap(existingConfig, proxyPath)
+	if err != nil {
+		return prevObservedProxyConfig, append(errs, err)
+	}
+
+	if len(currentProxyMap) > 0 {
+		unstructured.SetNestedStringMap(prevObservedProxyConfig, currentProxyMap, proxyPath)
+	}
+
+	observedConfig := map[string]interface{}{}
+	proxyConfig, err := listers.ProxyLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.Warningf("proxy.config.openshift.io/cluster: not found")
```

Comment:
this might be noisy, I would move it to V(4-5) ?

Reply:
check the service account one, use the same level

Reply:
The function used here still kept the `klog` but I removed this logging completely in the library-go PR